internal/app/init/internal/rootfs/rootfs.go

```diff
@@ -76,6 +84,36 @@ func Prepare(s string, data *userdata.UserData) (err error) {
 	return nil
 }
 
+func generatePKI(data *userdata.UserData) (err error) {
```

Comment:
I felt this was the most appropriate place to place the PKI logic since this package is responsible for writing the configuration to disk.